web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java

```diff
@@ -83,6 +89,33 @@ public void setConfiguration(RandomizationConfiguration configuration) {
         }
     }
 
+    public void refreshConfigurations(String accessToken) {
+        List<RandomizationConfiguration> randomizationConfigurations = new ArrayList<>();
+        ArrayList<Study> studies = studyDao.findAll();
+        studies.stream().filter(study->study.getStatus() == Status.AVAILABLE
+                && StringUtils.isNotEmpty(study.getStudyEnvUuid()))
+                .forEach(study-> {
+                    List<ModuleConfigDTO> configs = studyBuildService.getModuleConfigsFromStudyService(accessToken, study);
+                    String moduleEnabled = studyBuildService.isModuleEnabled(configs, study, Modules.RANDOMIZE);
+                    if (StringUtils.equalsIgnoreCase(moduleEnabled, StudyBuildService.ENABLED))
+                        try {
+                            randomizationConfigurations.add(retrieveConfiguration(study.getStudyEnvUuid(), accessToken));
+                        } catch (Exception e) {
```

Comment:
if there is an error retrieving configuration, should we be returning an error response instead of swallowing the exception?

Reply:
Logging the exception now.